verilog/tools/kythe/kythe_facts_extractor.cc

```diff
@@ -725,16 +692,18 @@ std::string GetFilePathFromRoot(const IndexingFactNode& root) {
   return root.Value().Anchors()[0].Value();
 }
 
-std::ostream& KytheFactsPrinter::Print(std::ostream& stream) const {
-  KytheFactsExtractor kythe_extractor(GetFilePathFromRoot(root_), &stream);
-
-  kythe_extractor.ExtractKytheFacts(root_);
-  return stream;
-}
+std::ostream& KytheFactsPrinter::Print(std::ostream& stream,
+                                       const IndexingFactNode& root) {
+  if (!files_scope_resolvers_.empty()) {
+    files_scope_resolvers_.push_back(
+        ScopeResolver(files_scope_resolvers_.back()));
+  } else {
+    files_scope_resolvers_.push_back(ScopeResolver(nullptr));
+  }
```

Comment:
I wouldn't expect the Print() function to alter the state of the object (unless it's related to printing).

Can you move this code to another function? Simply call the print for the whole tree and not file-by-file.

Reply:
This is not yet done. David made a similar comment.